### PR TITLE
Unittest to pytest

### PR DESCRIPTION
**Pull request by cl0cks4fe**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Glanced over this, and it looks pretty good. However, #32360 was opened first so @PZES has the lock on this bounty. If they give it up, you'll be next up for it.

**Comment by adeebshihadeh** [comment]
Ended up closing the original one. This looks a lot better than that one, but still has some unittest references:
```
batman:fourpilot$ rg unitt -l | grep -v tinygrad | grep -v opendbc | grep -v panda | grep -v cereal | grep -v teleoprtc | grep -v rednose_repo
poetry.lock
tools/car_porting/test_car_model.py
tools/lib/tests/test_logreader.py
system/webrtc/tests/test_webrtcd.py
system/webrtc/tests/test_stream_session.py
selfdrive/updated/tests/test_base.py
selfdrive/thermald/tests/test_fan_controller.py
selfdrive/thermald/tests/test_power_monitoring.py
selfdrive/athena/tests/test_registration.py
selfdrive/athena/tests/test_athenad_ping.py
selfdrive/athena/tests/test_athenad.py
selfdrive/car/tests/test_fw_fingerprint.py
selfdrive/car/tests/test_lateral_limits.py
selfdrive/ui/tests/test_ui/run.p
```

Let's also block unittest like Shane mentioned here: https://github.com/commaai/openpilot/pull/32360#issuecomment-2102051920

**Comment by cl0cks4fe** [comment]
@adeebshihadeh 
`selfdrive/ui/tests/test_ui/run.py` and `tools/car_porting/test_car_model.py` are both using unittest to run more as tools than as tests. can likely hack that around but didn't want to lump into this PR

**Comment by adeebshihadeh** [comment]
Can you rebase and mark as ready for review if those are the last two left?

RE: the last two, let's still add the ruff rule to error on `unittest` usage, then add an inline ignore in those files.

**Comment by cl0cks4fe** [comment]
happy to split this up or something if it that makes attacking it easier

what's left is: 
- `selfdrive/ui/tests/test_ui/run.py` (wip https://github.com/commaai/openpilot/pull/32432)
- `selfdrive/car/tests/test_models.py`, as it would break `tools/car_porting/test_car_model.py` (i have a working version of this working also removing the dep)

File: selfdrive/car/tests/test_lateral_limits.py [hunk 1]
```diff
@@ -66,26 +65,8 @@ def calculate_0_5s_jerk(control_params, torque_params):
   def test_jerk_limits(self):
     up_jerk, down_jerk = self.calculate_0_5s_jerk(self.control_params, self.torque_params)
     car_model_jerks[self.car_model] = {"up_jerk": up_jerk, "down_jerk": down_jerk}
-    self.assertLessEqual(up_jerk, MAX_LAT_JERK_UP + MAX_LAT_JERK_UP_TOLERANCE)
-    self.assertLessEqual(down_jerk, MAX_LAT_JERK_DOWN)
+    assert up_jerk <= MAX_LAT_JERK_UP + MAX_LAT_JERK_UP_TOLERANCE
+    assert down_jerk <= MAX_LAT_JERK_DOWN
 
   def test_max_lateral_accel(self):
```
Comment (sshane) [inline selfdrive/car/tests/test_lateral_limits.py:71]: @adeebshihadeh this report was useful

Comment (adeebshihadeh) [reply selfdrive/car/tests/test_lateral_limits.py:71]: my bad, didn't notice it got removed

Comment (cl0cks4fe) [reply selfdrive/car/tests/test_lateral_limits.py:71]: https://github.com/commaai/openpilot/pull/32614

**Pull request merged**